packages/marketplace-ui/src/components/MyTokens.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Box,
+  Button,
+  Container,
+  Center,
+  Stack,
+  Text,
+  Flex,
+  VStack,
+} from "@chakra-ui/react";
+import { PublicKey } from "@solana/web3.js";
+import React, { useCallback, useEffect, useState } from 'react';
+import { useWallet } from "@solana/wallet-adapter-react";
+import { useWalletModal } from "@solana/wallet-adapter-react-ui";
+import { useProvider } from "@strata-foundation/react";
+import { TOKEN_PROGRAM_ID } from "@solana/spl-token";
+import { TokenAccountParser } from "@strata-foundation/react";
+import { truthy } from "@strata-foundation/spl-utils";
+import { TokenItem } from "./TokenItem";
+import { MetadataData } from "@metaplex-foundation/mpl-token-metadata";
+
+export const MyTokens = () => {
+  const { publicKey, connected } = useWallet();
+  const { visible, setVisible } = useWalletModal();
+  const { provider } = useProvider();
+  const [mints, setMints] = useState<PublicKey[]>([]);
+  useEffect(() => {
+    async function getTokenAccounts() {
+      if (!publicKey || !provider) return;
+      const accounts = await provider.connection.getTokenAccountsByOwner(publicKey, { programId: TOKEN_PROGRAM_ID });
+      const m = accounts.value.map((acc) => (TokenAccountParser(acc.pubkey, acc.account)?.info.mint)).filter(truthy);
+      setMints(m);
+    }
+    getTokenAccounts();
+  }, [publicKey, provider]);
+
+
+  const [tokensRecord, setTokensRecord] = useState<Record<string,Set<string>>>({});
+  // TODO this intermediate token detection is flawed, if bonding curve is closed no way to detect which is the real token to display
```

Comment:
Is this still an issue? Thought we were going to close the ata of the intermediate token?

Reply:
True, deleted

Reply:
Actually it's probably worth keeping this logic for now to detect ATAs that haven't been closed yet